tests/src/GC/Performance/Tests/LowLatencyTest.cs

```diff
@@ -0,0 +1,185 @@
+                    break;
+                case "batch":
+                    gcMode = GCLatencyMode.Batch;
+                    break;
+                default:
+                    Usage();
+                    return;
+            }
+
+            GCTestC test = new GCTestC();
+            test.DoTest(iters, gcMode);
+        }
+
+
+        private void LoadData( int count, ref long[] values )
+        {
+
+            loid++;
+            Hashtable aMap = new Hashtable(count);
+            byte[] aBuffer = null;
+
+            using (StreamReader reader = new StreamReader("clunie_small.xml"))
+            {
+                aBuffer = new byte[reader.BaseStream.Length];
+                reader.BaseStream.Read(aBuffer, 0, (int)reader.BaseStream.Length);
+            }
+
+            using (MemoryStream aMemStream = new MemoryStream(aBuffer))
+            {
+                aBuffer = null;
+
+                values[0] = PerformanceCounter.QueryCounter;
+                for (int i = 0; i < count; i++)
+                {
+                    Thread.Sleep(0); // simulate waiting on arrival of new data...
+
+                    // create XmlDocuments
+                    XmlDocument aXmlDoc = new XmlDocument();
+                    aXmlDoc.Load(aMemStream);
+                    aMap.Add("1.2.3" + loid.ToString() + i.ToString(), aXmlDoc);
+                    aXmlDoc = null;
+
+                    // reset the position in the memory stream to the beginning
+                    aMemStream.Seek(0, SeekOrigin.Begin);
+
+                    values[i + 1] = PerformanceCounter.QueryCounter;
+                }
+
+            }
+
+        }
+
+
+        private void PrintStat(long[] values)
+        {
+            int count=values.Length-1;
+            long max = values[1]-values[0];
+
+            for (int i=0;i<count;i++)
+            {
+                max = Math.Max(max, (values[i+1]-values[i]));
+            }
+
+            //Console.WriteLine("Maximum of {0}: {1}", count, PerformanceCounter.PrintTime(PerformanceCounter.Seconds(max)));
+
+        }
+
+
+        public void DoTest(int count, GCLatencyMode gcMode)
+        {
+
+            GCLatencyMode oldMode = GCSettings.LatencyMode;
+
+            long[] values = new long[count+1]; // reuse this array for every test ...
+
+            RuntimeHelpers.PrepareConstrainedRegions();
+            try
+            {
+                // Load Data
+                GCSettings.LatencyMode = gcMode;
+                LoadData(count, ref values);
+            }
+            finally
+            {
+                GCSettings.LatencyMode = oldMode;
+            }
+
+            PrintStat(values);
+        }
+
+
+        public static class PerformanceCounter
+        {
+            [DllImport("kernel32.dll", CallingConvention=CallingConvention.Winapi, EntryPoint="QueryPerformanceFrequency")]
+            private static extern int QueryPerformanceFrequency_int(ref long count);
```

Comment:
Can you see if you can replace these p/invokes into kernel32 with cross-platform implementations (e.g. System.Diagnostics.Stopwatch)? It seems like Stopwatch could be a good replacement for this PerformanceCounter class.